.github/workflows/java.yml

```diff
@@ -2,26 +2,98 @@ name: Java
 
 on:
   push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+    branches:
+      - '**'
+    tags-ignore:
+      - '**'
 
 jobs:
-  test:
-    name: Test
+  build-and-test:
+    name: Build and Test
     strategy:
+      fail-fast: false
       matrix:
         os: [ ubuntu-latest ]
-        java: [ 8, 11, 17 ]
+        java: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17 ]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout current branch
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Setup JDK
         uses: actions/setup-java@v2
         with:
           java-version: ${{ matrix.java }}
           distribution: 'adopt'
           cache: gradle
+      - name: Validate gradle wrapper
+        uses: gradle/wrapper-validation-action@v1.0.4
       - name: Test
         run: cd java && ./gradlew test
+
+  set-release-version:
+    name: Prepare version to tag and release
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2 #we need to make a git diff of the last 2 commits
+      - name: Get project version
+        id: get_project_version
+        run: |
```

Comment:
As discussed, this is a different approach compared to the Java SDK but we agreed that's fine having projects-tailored approaches that diverge a bit